Example/Tests/BONUtilitiesTestCase.m

```diff
@@ -0,0 +1,252 @@
+        },
+    };
+
+    BONAssertAttributedStringHasAttributes(textView.attributedText, controlAttributes);
+}
+
+#pragma mark - UITextField
+
+- (void)testTextFieldTextAfterChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textField.textAndApplyChainable = @"Hello, world!";
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldTextBeforeChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.textAndApplyChainable = @"Hello, world!";
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldAttributedTextAfterChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+    textField.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+- (void)testTextFieldAttributedTextBeforeChainable
+{
+    UITextField *textField = UITextField.new;
+    textField.attributedText = [[NSAttributedString alloc] initWithString:@"Hello, world!"];
+    textField.bonChainable = BONChain.new.font([UIFont systemFontOfSize:16]);
+
+    XCTAssertNotNil(textField.bonChainable);
+    XCTAssertEqualObjects(textField.attributedText.string, @"Hello, world!");
+
+    NSMutableDictionary *controlAttributes = [self defaultAttributesForClass:[UITextField class] withString:@"Hello, world!"];
+    [controlAttributes[BONValueFromRange(0, 13)] addEntriesFromDictionary:@{
+        NSFontAttributeName : [UIFont systemFontOfSize:16],
+    }];
+
+    BONAssertAttributedStringHasAttributes(textField.attributedText, controlAttributes);
+}
+
+#pragma mark - Helpers
+
+- (NSMutableDictionary *)defaultAttributesForClass:(Class)viewClass withString:(NSString *)string
```

Comment:
#pp I prefer to capitalize the names of `Class` parameters: `defaultAttributesForClass:(Class)ViewClass`. It makes it look better (IMHO) when you call methods on it - easier to see that that's a class method.